### PR TITLE
Add mcversion to PluginMetadata

### PR DESCRIPTION
**Pull request by windy1**
Signed-off-by: Walker Crouse <walkercrouse@hotmail.com>

### CODE REVIEW

**Comment by Minecrell** [comment]
When and where would that be set? It doesn't make much sense for the plugin annotation, so the only way would be to add it manually to `mcmod.info` file.

**Comment by windy1** [comment]
It's in the spec and is used for forge mods, although plugin's are supposed to be MC version independent, it's nice information to have (https://github.com/SpongePowered/Ore/issues/118). Additionally, maybe the annotation processor could infer the correct MC version from the Sponge API version a plugin is using, but that's beyond the scope of this PR.

File: src/main/java/org/spongepowered/plugin/meta/PluginMetadata.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
 
     @Nullable private String description;
     @Nullable private String url;
+    @Nullable private String mcVersion;
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/plugin/meta/PluginMetadata.java]: Can you change this to `minecraftVersion`? The field names do not strictly follow the mcmod.info naming

**Comment by windy1** [comment]
Updated @Minecrell

**Comment by Minecrell** [comment]
:+1:

**Comment by Minecrell** [comment]
I added https://github.com/SpongePowered/SpongeGradle/commit/937f0d8edaa43fa706262c10ff4f760a5d3582ed which should automatically set this when using ForgeGradle together with SpongeGradle.

**Pull request merged**